common/bag_time_manager_rviz_plugin/src/bag_time_manager_panel.cpp

```diff
@@ -66,12 +68,17 @@ BagTimeManagerPanel::BagTimeManagerPanel(QWidget * parent) : rviz_common::Panel(
 void BagTimeManagerPanel::onInitialize()
 {
   raw_node_ = this->getDisplayContext()->getRosNodeAbstraction().lock()->get_raw_node();
-  client_pause_ =
-    raw_node_->create_client<Pause>("/rosbag2_player/pause", rmw_qos_profile_services_default);
-  client_resume_ =
-    raw_node_->create_client<Resume>("/rosbag2_player/resume", rmw_qos_profile_services_default);
-  client_set_rate_ =
-    raw_node_->create_client<SetRate>("/rosbag2_player/set_rate", rmw_qos_profile_services_default);
+
+// APIs taking rclcpp::QoS objects are only available in ROS 2 Iron and higher
+#if RCLCPP_VERSION_MAJOR >= 18
```

Comment:
As with switching to galactic, using `ROS_DISTRO_ROLLING` is more useful for maintenance.

Reply:
`ROS_DISTRO_ROLLING` does not point to a specific release, but to the latest commit of the ROS repositories, so if I use `ROS_DISTRO_ROLLING` and once `iron` is released, this code would become invalid. It's more precise to use the version in a package dependency, as it'll be more correct no matter what happens to `rolling`.

Reply:
It's the same with `RCLCPP_VERSION_MAJOR >= 18`. It needs to be `==` to use a specific release.

Reply:
It's different, with `>=` we make sure that it'll work with `iron` and higher, if we only check `rolling`, then when `iron` is released, the code will fail because `iron` will no longer be the same as `rolling` and we will have to update the code. With this change, nothing needs to be updated when `iron` is released.

Reply:
There is no guarantee that this interface will work with rclcpp version 19. Both `RCLCPP_VERSION_MAJOR >= 18` and `ROS_DISTRO_ROLLING` can error when the code is updated.

Reply:
Actually this is a small problem, so we should consider it when we support iron.